### PR TITLE
Enable use of c27-c31 by userland.

### PR DESCRIPTION
**Pull request by bsdjhb**
With these changes, the kernel still has to avoid the use of $c27 in C code since nested traps assume they can use C27.  I could perhaps fix this by only using kr1c to save $c27 for short periods of time when $c27 is actually used.

### CODE REVIEW

File: sys/mips/mips/exception.S [hunk 1]
```diff
@@ -95,7 +95,7 @@
 #ifdef CPU_CHERI
 #include <machine/cheriasm.h>
 #include <machine/cherireg.h>
-.set cheri_sysregs_accessible	# Don't warn on access to c27-c30
+.set cheri_sysregs_accessible	# Don't warn on access to c27-c30 ???
```
Comment (bsdjhb) [inline sys/mips/mips/exception.S]: The ??? is that I'm not sure if this is still relevant.

Comment (arichardson) [reply sys/mips/mips/exception.S]: I think it's been long enough to make the the default and remove the option.

**Comment by bsdjhb** [comment]
Yeah, I agree the KR1C name for the GPR is confusing.  I could maybe name the aliases something like KSCRATCH1 and KSCRATCH2 instead?

**Comment by arichardson** [comment]
Yes, I think that sounds good.

**Comment by jrtc27** [comment]
Hm, this seems sensible, other than needing to avoid C27 in generated kernel code (and thus requiring an LLVM flag for whether to allocate it or not). Is it awkward to fix that?

It also looks like KSCRATCH2 (ie C28) is only used in locore; do we really need to give it a special reserved name everywhere, rather than just use any appropriate capability register (which can be C28) inside locore, as you would for any other register allocation in assembly?

**Comment by bsdjhb** [comment]
I would be fine dropping the KSCRATCH2 name entirely.  I do think I can fix the C27 issue but probably want to push what is there now and then do that as a followup.  The way I would fix C27 would be that I would explicitly stash it in KR1C and then restore it again only when I need to use it as a scratch.  This will add a few more instructions though, but I think it is cleaner overall.  Mostly it means auditing all the places that use C27 currently and have them either explicitly stash it in KR1C before and after, or see if it's in a place where C27 is free to reuse (an example of the latter would be to make sure C27 is saved in the trapframe first before saving any of the special registers like DDC, probably by moving save of DDC, etc. to the end of the relevant macros).

**Comment by bsdjhb** [comment]
Fixing C27 was easier than I had feared, so this includes that now as well.

**Pull request merged**